src/main/kotlin/com/epam/brn/model/Exercise.kt

```diff
@@ -1,21 +1,67 @@
 package com.epam.brn.model
 
+import com.epam.brn.dto.ExerciseDto
 import javax.persistence.Entity
 import javax.persistence.GeneratedValue
 import javax.persistence.GenerationType
 import javax.persistence.Id
 import javax.persistence.JoinColumn
+import javax.persistence.SequenceGenerator
+import javax.persistence.CascadeType
 import javax.persistence.ManyToOne
+import javax.persistence.OneToMany
 
 @Entity
 data class Exercise(
     @Id
-    @GeneratedValue(strategy = GenerationType.IDENTITY)
-    val id: Long = 0,
+    @GeneratedValue(generator = "exercise_id_seq", strategy = GenerationType.SEQUENCE)
+    @SequenceGenerator(
+        name = "exercise_id_seq",
+        sequenceName = "exercise_id_seq",
+        allocationSize = 50
+    )
+    val id: Long? = null,
     val name: String,
     val description: String? = "",
     val level: Short? = 0,
     @ManyToOne
     @JoinColumn(name = "exercise_series_id")
-    var exerciseSeries: ExerciseSeries
-)
\ No newline at end of file
+    var series: Series,
+    @OneToMany(mappedBy = "exercise", cascade = [CascadeType.ALL])
+    val tasks: MutableSet<Task> = HashSet()
+) {
+    fun toDto() = ExerciseDto(
+        id = id,
+        name = name,
+        description = description,
+        tasks = tasks.map { task -> task.toDto() }.toMutableSet()
+    )
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
What do you think about using somthing like lombok annotation @EqualsAndHashcode for generating this methods?